react/src/components/timeline-app/TimelineComponent.jsx

```diff
@@ -0,0 +1,177 @@
+          second: 'ddd D MMMM YYYY',
+          minute: 'ddd D MMMM YYYY',
+          hour: 'ddd D MMMM YYYY',
+          weekday: 'MMMM YYYY',
+          day: 'MMMM YYYY',
+          month: 'YYYY',
+          year: '',
+        },
+      },
+      maxHeight: '25em',
+      minHeight: '10em',
+      orientation: 'top',
+      selectable: false,
+      showCurrentTime: false,
+      template: (event, element) => {
+        const className = 'timeline__itembox ' +
+          (event.icon ? 'timeline__itembox--hasicon ' : '') +
+          (event.linkType === 'external' ? 'timeline__itembox--external ' : '');
+        const html = (
+          <div>
+            <div className={ className } id={ 'timeline__itembox__' + event.id }>
+              { event.icon ? <div className="timeline__itembox__imagebox"><img src={ event.icon } /></div> : '' }
+              <div className="timeline__itembox__titlebox">
+                {
+                  event.link ?
+                    <a href={ event.link } target={ event.linkType === 'external' ? '_blank' : '_self' }>
+                      { event.title }
+                    </a>
+                    : event.title
+                }
+              </div>
+            </div>
+          </div>
+        );
+        return ReactDOM.render(html, element);
+      },
+      zoomMin: 1000 * 60 * 60 * 24 * 3,
+      zoomMax: 1000 * 60 * 60 * 24 * 365 * (margin + 10) * 2,
+    };
+    const timeline = new vis.Timeline(container, items, options);
+
+    /* Create the tooltips */
+    events.forEach((event) => {
+      const date = event.start;
+      let tooltipContent = `<p class="timeline__tooltip__date">${date.getDate()}.${date.getMonth() + 1}.${date.getFullYear()}</p>`;
+      if (event.description) {
+        tooltipContent += event.description;
+      }
+      tooltipContent += `<span class="timeline__tooltip__arrow"></span>`;
+      const tooltipNode = document.createElement('div');
+      tooltipNode.innerHTML = tooltipContent;
+      tooltipNode.classList.add('timeline__tooltip');
+      tooltipNode.id = 'timeline__tooltip__' + event.id;
+      container.appendChild(tooltipNode);
+    });
+    // The event 'itemover' cannot be used because the original event will not be passed due to a bug.
+    // It is already fixed by the developers but not included in the current release (4.18.1).
+    // https://github.com/almende/vis/pull/2704
+    let currentItem = null;
+    timeline.on('mouseOver', (e) => {
+      if (e.what === 'item' && currentItem !== e.item) {
+        const tooltipNode = document.getElementById('timeline__tooltip__' + e.item);
+        currentItem = e.item;
+        if (tooltipNode) {
+          const margin = 20;
+          const tooltipWidth = tooltipNode.offsetWidth;
+          const itemBoundaries =
+            document.getElementById('timeline__itembox__' + e.item).parentNode.parentNode.parentNode.getBoundingClientRect();
+          const mouseX = e.pageX;
+          let left = mouseX;
```

Comment:
getting a little bit difficult for me to track the logic, but consider doing this with react components, should be possible?


I'll stop here with the review, maybe we do one more iteration after you push the changes.